File: VB/LabelsOverlap/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraCharts
' ...

Namespace LabelsOverlap
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub comboBoxEdit1_Properties_SelectedValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles comboBoxEdit1.Properties.SelectedValueChanged
			Select Case comboBoxEdit1.SelectedIndex
				Case 0
					SetOverlappingMode(True, ResolveOverlappingMode.Default)
				Case 1
					SetOverlappingMode(True, ResolveOverlappingMode.HideOverlapped)
				Case 2
					SetOverlappingMode(True, ResolveOverlappingMode.JustifyAroundPoint)
				Case 3
					SetOverlappingMode(True, ResolveOverlappingMode.JustifyAllAroundPoint)
				Case 4
					SetOverlappingMode(False, ResolveOverlappingMode.Default)
			End Select
		End Sub

		Private Sub SetOverlappingMode(ByVal resolve As Boolean, ByVal mode As ResolveOverlappingMode)
			Dim label As SeriesLabelBase = chartControl1.Series(0).Label
			label.OverlappingOptions.ResolveOverlapping = resolve
			If resolve Then
				label.OverlappingOptions.Mode = mode
			End If
		End Sub

	End Class
End Namespace

File: CS/LabelsOverlap/Form1.cs
using System;
using System.Windows.Forms;
using DevExpress.XtraCharts;
// ...

namespace LabelsOverlap {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void comboBoxEdit1_Properties_SelectedValueChanged(object sender, EventArgs e) {
            switch (comboBoxEdit1.SelectedIndex) {
                case 0:
                    SetOverlappingMode(ResolveOverlappingMode.Default);
                    break;
                case 1:
                    SetOverlappingMode(ResolveOverlappingMode.HideOverlapped);
                    break;
                case 2:
                    SetOverlappingMode(ResolveOverlappingMode.JustifyAroundPoint);
                    break;
                case 3:
                    SetOverlappingMode(ResolveOverlappingMode.JustifyAllAroundPoint);
                    break;
                case 4:
                    SetOverlappingMode(ResolveOverlappingMode.Default);
                    break;
            }
        }

        private void SetOverlappingMode(ResolveOverlappingMode mode) {
            chartControl1.Series[0].Label.ResolveOverlappingMode = mode;
        }

    }
}